tests/queries/0_stateless/02302_s3_file_pruning.sql

```diff
@@ -5,7 +5,9 @@
 drop table if exists test_02302;
 create table test_02302 (a UInt64) engine = S3(s3_conn, filename='test_02302_{_partition_id}', format=Parquet) partition by a;
 insert into test_02302 select number from numbers(10) settings s3_truncate_on_insert=1;
-select * from test_02302;  -- { serverError 48 }
+select * from test_02302 order by a;
```

Comment:
> Run before this select
> 
> ```
> create table test_02302_another (a UInt64) engine = S3(s3_conn, filename='test_02302_{_partition_id}_another', format=Parquet) partition by a;
> insert into test_02302_another select number + 10 from numbers(10) settings s3_truncate_on_insert=1;
> ```
> 
> And you'll see how select `test_02302` and `test_2302_another` will read files from both which is wrong for partition wildcard.

I have modified my code.

Reply:
As I said it's not 100% correct that you get all the files so the solution is wrong.
`partition_id` is a way to make **writes** work for file-based storages like S3, URL, File but for reads, it's not clear what is correct.
What you can do to get the same behavior is creating one table for inserting with `file_{_partition_id}` and another table which reads from `file_*`.
In no other engines do we do something like this and I think it can lead to strange results.